server/mailrepository/mailrepository-api/src/test/java/org/apache/james/mailrepository/api/properties/eventsourcing/PropertiesAggregateIdTest.java

```diff
@@ -0,0 +1,77 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailrepository.api.properties.eventsourcing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.apache.james.mailrepository.api.MailRepositoryUrl;
+import org.junit.jupiter.api.Test;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+class PropertiesAggregateIdTest {
+
+    @Test
+    void shouldMatchBeanContract() {
+        EqualsVerifier.forClass(PropertiesAggregateId.class)
+            .verify();
+    }
+
+    @Test
+    void constructorShouldThrowWhenPassingNullUrl() {
+        assertThatThrownBy(() -> new PropertiesAggregateId(null))
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void parseShouldThrowWhenAggregateIdStringIsNull() {
+        assertThatThrownBy(() -> PropertiesAggregateId.parse(null))
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void parseShouldThrowWhenAggregateKeyStringDoesntStartWithASlash() {
+        assertThatThrownBy(() -> PropertiesAggregateId.parse("MailRepositoryProperties/protocol://var/mail"))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("aggregate key string should start with '/MailRepositoryProperties'");
+    }
+
+    @Test
+    void parseShouldThrowWhenAggregateKeyStringDoesntContainsTheRightAggregatePrefixAfterTheSlash() {
+        assertThatThrownBy(() -> PropertiesAggregateId.parse("/AnotherAggregateKey/protocol://var/mail"))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("aggregate key string should start with '/MailRepositoryProperties'");
+    }
+
+    @Test
+    void parseShouldThrowWhenAggregateKeyStringDoesntContainsTheRightMailRepositoryUrlFormat() {
+        assertThatThrownBy(() -> PropertiesAggregateId.parse("/MailRepositoryProperties/protocol:var/mail"))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void parseShouldReturnTheCorrespondingAggregateId() {
+        String mailRepositoryUrlString = "protocol://var/mail";
+
+        assertThat(PropertiesAggregateId.parse("/MailRepositoryProperties" + mailRepositoryUrlString))
+            .isEqualTo(new PropertiesAggregateId(MailRepositoryUrl.from(mailRepositoryUrlString)));
+    }
```

Comment:
What happens when the protocol is invalid?

Reply:
there is a test above `parseShouldThrowWhenAggregateKeyStringDoesntContainsTheRightMailRepositoryUrlFormat`